### PR TITLE
Resync v0.0.10 of service-catalog with origin

### PR DESCRIPTION
**Pull request by jpeeler**
This specifically uses a different branch than master as suggested to see if verify-upstream-commits.sh will be happy.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Unclear why that commit is being flagged as an `UPSTREAM`... [`Commit.DeclaresUpstreamChange()](https://github.com/openshift/origin/blob/master/tools/rebasehelpers/util/git.go#L39-L41) doesn't apply:

```go
func (c Commit) DeclaresUpstreamChange() bool {
	return strings.HasPrefix(strings.ToLower(c.Summary), "upstream")
}
```

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
flake #8502

**Comment by jpeeler** [comment]
[test] again, the previous commit was just some additional error checking.

File: tools/rebasehelpers/util/git.go [hunk 1]
```diff
@@ -335,6 +341,22 @@ func filesInCommit(sha string) ([]File, error) {
 	return files, nil
 }
 
+func descriptionInCommit(sha string) ([]string, error) {
+	descriptionLines := []string{}
+	stdout, stderr, err := run("git", "show", "--quiet", sha)
```
Comment (stevekuznetsov) [inline tools/rebasehelpers/util/git.go]: Why not `git log --pretty=%b -1 $SHA`?

Comment (jpeeler) [inline tools/rebasehelpers/util/git.go]: That is a bit nicer. If/when CI fails, I'll incorporate it.

**Comment by stevekuznetsov** [comment]
So what was the root cause? I'm missing something because I can't tell how your commit got picked for the `UPSTREAM` validation without having that as the prefix for your subject

**Comment by jpeeler** [comment]
@stevekuznetsov Due to the way subtrees work, the commit looked like it was an origin vendor change. The sc vendor directory prefix is not present in the file list, though it is in the commit description.

**Comment by stevekuznetsov** [comment]
Ah, I see

**Comment by derekwaynecarr** [comment]
[merge][severity:blocker]

**Comment by derekwaynecarr** [comment]
@eparis @smarterclayton - green if willing to manual merge.

File: tools/rebasehelpers/util/git.go (hunk 1 above)
Comment (stevekuznetsov) [inline tools/rebasehelpers/util/git.go]: @jpeeler can you open a follow-up pr for this please

**Comment by eparis** [comment]
Can't break anything but itself. Merging to keep the queue open for meaningful things.

File: tools/rebasehelpers/util/git.go (hunk 1 above)
Comment (jpeeler) [inline tools/rebasehelpers/util/git.go]: https://github.com/openshift/origin/pull/14712

**Pull request merged**